### DISCUSSION TITLE
HTTPSRedirectMiddleware causing too many redirects (and system doesn't work)

### DISCUSSION THREAD

**Discussion by tomazburd** [fastapi/fastapi#11438] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.middleware.httpsredirect import HTTPSRedirectMiddleware

app = FastAPI()

app.add_middleware(HTTPSRedirectMiddleware)


@app.get("/")
async def main():
    return {"message": "Hello World"}
```

### Description

I set up a server with a domain name on Nginx server. I used Letsencrypt. When I apply the code, I get too many redirect failures and I don't understand why.

If I don't use HTTPSRedirectMiddleware, than all the requests from HTTP are ofcourse not redirected and user gets 404. Is there a way to fix that?

### Operating System

Windows

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

3.10

**Comment by sehraramiz** [comment]
have you considered to config nginx to redirect http to https?
like this
```nginx
server {
    listen 80 default_server;
    server_name _;
    return 301 https://$host$request_uri;
}
```
can you share a snippet of your nginx config?

**Comment by YuriiMotov** [comment]
Looks like the author is not interested in this anymore. 
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
